app/src/main/java/com/ojasvi/memify/ui/screens/PhotoReasoningViewModel.kt

```diff
@@ -97,10 +129,31 @@ class PhotoReasoningViewModel(
                     .joinToString("") { it.text.toString() } // Joins responses into a single string
 
                 // Update the UI state with the final output
-                _uiState.value = PhotoReasoningUiState.Success(outputContent)
+                _uiState.value = PhotoReasoningUiState.GeminiSuccess(outputContent)
             } catch (e: Exception) {
                 _uiState.value = PhotoReasoningUiState.Error(e.localizedMessage ?: "")
             }
         }
     }
+
+    private fun generateImage(prompt: String, onImageUrlGenerated: (String?) -> Unit) {
+        val request = ImageRequest(prompt)
+        RetrofitInstance.api.generateImage(request).enqueue(object : Callback<ImageResponse> {
+            override fun onResponse(call: Call<ImageResponse>, response: Response<ImageResponse>) {
+                if (response.isSuccessful && response.body() != null) {
+                    val imageUrl = response.body()!!.image_url
+                    Log.d("ImageGeneration", "Image URL: $imageUrl")
+                    onImageUrlGenerated(imageUrl) // Pass the image URL to the callback
+                } else {
+                    Log.e("ImageGeneration", "Error: ${response.code()}")
+                    onImageUrlGenerated(null) // Pass null if there's an error
+                }
+            }
+
+            override fun onFailure(call: Call<ImageResponse>, t: Throwable) {
+                Log.e("ImageGeneration", "Failure: ${t.message}")
+                onImageUrlGenerated(null) // Pass null on failure
+            }
+        })
+    }
```

Comment:
This should be a suspend function in the API interface.

Reply:
Noted. Will do this after completing the work on basic UI fixes.